### PR TITLE
Add calledOnceWith and calledOnceWithExactly

### LINKED ISSUE #116: Support calledOnceWithExactly and calledOnceWith
Sinon was released in version [4.3.0](https://github.com/sinonjs/sinon/releases/tag/v4.3.0) with `calledOnceWithExactly` and `calledOnceWith`. Would be great to support these methods which will lead to less boilerplate if testing a call happened once and with an exact argument.

https://github.com/sinonjs/sinon/pull/1686
https://github.com/sinonjs/sinon/issues/1247

### PR DESCRIPTION
**Pull request by JayAndCatchFire**
This should resolve #116.

### CODE REVIEW

File: lib/sinon-chai.js [hunk 1]
```diff
@@ -140,7 +140,9 @@
     sinonMethod("calledImmediatelyAfter", "been called immediately after %1");
     sinonMethod("calledOn", "been called with %1 as this", ", but it was called with %t instead");
     sinonMethod("calledWith", "been called with arguments %*", "%D");
+    sinonMethod("calledOnceWith", "been called exactly once with arguments %*", "%C");
     sinonMethod("calledWithExactly", "been called with exact arguments %*", "%D");
+    sinonMethod("calledOnceWithExactly", "been called exactly once with exact arguments %*", "%C");
```
Comment (JayAndCatchFire) [inline lib/sinon-chai.js]: So, here's the thing about these, specifically their `nonNegatedSuffix`es.  There's really no good way to address these new ones.

Using the `", but it was called %c%C"` suffix doesn't make sense when it's only been called once.  And using `"%D"` causes some odd formatting issues in the case of multiple calls.

With `"%D"`, it ends up looking like this:
```
     AssertionError: expected spy to have been called with exact arguments A, BCall 1:
C A
A B
Call 2:
C A
B
```

I'm about 99.999% sure this is a sinon formatter issue and not an issue with sinon-chai.

Comment (JayAndCatchFire) [reply lib/sinon-chai.js]: This should be resolved by this ticket: https://github.com/sinonjs/sinon/issues/1717

I'm still open to suggestions on how these `nonNegatedSuffix`es should be.

Comment (domenic) [reply lib/sinon-chai.js]: I think just using `"%D"` should work pretty well, after that Sinon bug is fixed at least. Do you agree?

File: package.json [hunk 2]
```diff
@@ -32,11 +32,11 @@
     "sinon": "^1.4.0 || ^2.1.0 || ^3.0.0 || ^4.0.0"
   },
   "devDependencies": {
-    "chai": "^4.1.0",
-    "eslint": "^3.19.0",
+    "chai": "^4.1.2",
+    "eslint": "^4.18.2",
```
Comment (domenic) [inline package.json]: Let's leave dev dependency updates for a separate PR, and only update Sinon in this PR.

**Comment by domenic** [comment]
LGTM, but can you regenerate package-lock.json? I don't think you rolled back the changes there.

**Comment by JayAndCatchFire** [comment]
Good call.  I did it all via Github.

**Comment by JayAndCatchFire** [comment]
What do you mean by "messages tests"?  As in testing the formatting of the failure messages?

**Comment by domenic** [comment]
Yeah, see messages.js. Historically we've had a lot of regressions there (e.g. via Sinon upgrades that make our current messages nonsensical) that those tests have caught.

**Comment by JayAndCatchFire** [comment]
I'll take a crack at that in the next couple of days when I get time.  Thanks for reviewing this.

**Comment by JayAndCatchFire** [comment]
@domenic Just added some messages tests.  Can you please take another look?

**Comment by domenic** [comment]
Sorry for the massive delay here :(. This will go out in a release today!
